lib/polymer_interop.dart

```diff
@@ -20,98 +20,236 @@ class Polymer {
   static String camelToDashCase(String camel) =>
       (_CaseMap['camelToDashCase'] as JsFunction).apply([camel]);
 
+  /// Polymer provides a custom API for manipulating DOM such that local DOM and
+  /// light DOM trees are properly maintained. These methods and properties have
+  /// the same signatures as their standard DOM equivalents, except that properties
+  /// and methods that return a list of nodes return an Array, not a NodeList.
+  ///
+  /// Note: All DOM manipulation must use this API, as opposed to DOM API directly
+  /// on nodes.
+  ///
+  /// Using these node mutation APIs when manipulating children ensures that
+  /// shady DOM can distribute content elements dynamically. If you change
+  /// attributes or classes that could affect distribution without using the
+  /// Polymer.dom API, call distributeContent on the host element to force it
+  /// to update its distribution.
+  ///
+  /// Other resources: https://www.polymer-project.org/1.0/docs/devguide/local-dom.html
   static PolymerDom dom(node) => new PolymerDom(node);
 }
 
-/// Proxy for https://www.polymer-project.org/1.0/docs/devguide/local-dom.html.
+/// Polymer provides a custom API for manipulating DOM such that local DOM and
+/// light DOM trees are properly maintained. These methods and properties have
+/// the same signatures as their standard DOM equivalents, except that properties
+/// and methods that return a list of nodes return an Array, not a NodeList.
+///
+/// Note: All DOM manipulation must use this API, as opposed to DOM API directly
+/// on nodes.
+///
+/// Using these node mutation APIs when manipulating children ensures that
+/// shady DOM can distribute content elements dynamically. If you change
+/// attributes or classes that could affect distribution without using the
+/// Polymer.dom API, call distributeContent on the host element to force it
+/// to update its distribution.
+///
+/// Other resources: https://www.polymer-project.org/1.0/docs/devguide/local-dom.html
 class PolymerDom {
-  JsObject _proxy;
+  final JsObject _proxy;
 
+  final Node node;
+
+  /// The insert, append, and remove operations are transacted lazily in certain
+  /// cases for performance. In order to interrogate the dom (e.g. offsetHeight,
+  /// getComputedStyle, etc.) immediately after one of these operations, call
+  /// PolymerDom.flush() first.
   static void flush() => _Polymer['dom'].callMethod('flush');
 
-  PolymerDom(node) : _proxy = _Polymer.callMethod(
-      'dom', [node is Node ? new JsObject.fromBrowserObject(node) : node]);
```

Comment:
I just finished testing everything and it seems to work fine, I think its safe to assume this was cruft code and we no longer need it.